lib/stitches/guided_fill.py

```diff
@@ -0,0 +1,355 @@
+    #   <bound method STRtree.__del__ of <shapely.strtree.STRtree instance at 0x0D2BFD50>> ignored
+    del strtree
+
+
+def stitch_line(stitches, stitching_direction, geometry, projected_points, max_stitch_length, min_stitch_length, row_spacing, skip_last, offset_by_half):
+    if stitching_direction == 1:
+        stitched_line, _ = raster_line_string_with_priority_points(
+            geometry, 0.0, geometry.length, max_stitch_length, min_stitch_length, projected_points, abs(row_spacing), offset_by_half, True)
+    else:
+        stitched_line, _ = raster_line_string_with_priority_points(
+            geometry, geometry.length, 0.0, max_stitch_length, min_stitch_length, projected_points, abs(row_spacing), offset_by_half, True)
+
+    stitches.append(Stitch(*stitched_line[0], tags=('fill_row_start',)))
+    for i in range(1, len(stitched_line)-1):
+        stitches.append(Stitch(*stitched_line[i], tags=('fill_row')))
+
+    if not skip_last:
+        if stitching_direction == 1:
+            stitches.append(
+                Stitch(*geometry.coords[-1], tags=('fill_row_end',)))
+        else:
+            stitches.append(
+                Stitch(*geometry.coords[0], tags=('fill_row_end',)))
+
+
+@debug.time
+def path_to_stitches(path, travel_graph, fill_stitch_graph, angle, row_spacing, max_stitch_length, min_stitch_length,
+                     running_stitch_length, skip_last, offset_by_half):
+    path = collapse_sequential_outline_edges(path)
+
+    stitches = []
+
+    # If the very first stitch is travel, we'll omit it in travel(), so add it here.
+    if not path[0].is_segment():
+        stitches.append(Stitch(*path[0].nodes[0]))
+
+    for edge in path:
+        if edge.is_segment():
+            new_stitches = []
+            current_edge = fill_stitch_graph[edge[0]][edge[-1]]['segment']
+            path_geometry = current_edge['geometry']
+            projected_points = current_edge['projected_points']
+            stitching_direction = 1
+            if (abs(edge[0][0]-path_geometry.coords[0][0])+abs(edge[0][1]-path_geometry.coords[0][1]) >
+                    abs(edge[0][0]-path_geometry.coords[-1][0])+abs(edge[0][1]-path_geometry.coords[-1][1])):
+                stitching_direction = -1
+            stitch_line(new_stitches, stitching_direction, path_geometry, projected_points,
+                        max_stitch_length, min_stitch_length, row_spacing, skip_last, offset_by_half)
+            current_edge['already_rastered'] = True
+            transfer_points_to_surrounding_graph(
+                fill_stitch_graph, current_edge, row_spacing, False, new_stitches, overnext_neighbor=True)
+            transfer_points_to_surrounding_graph(fill_stitch_graph, current_edge, row_spacing, offset_by_half,
+                                                 new_stitches, overnext_neighbor=False, transfer_forbidden_points=offset_by_half)
+
+            stitches.extend(new_stitches)
```

Comment:
There's a critical piece missing here.  After stitching a line, we need to remove all of the edges in the travel graph that the stitched line touched.  Without that, travel stitches can go on top of lines we've already stitched -- I've seen it happen in my tests.